src/core/ActionAtomistic.cpp

```diff
@@ -432,10 +435,22 @@ unsigned ActionAtomistic::getTotAtoms()const {
 }
 
 void ActionAtomistic::makeWhole() {
-  for(unsigned j=0; j<positions.size()-1; ++j) {
-    const Vector & first (positions[j]);
-    Vector & second (positions[j+1]);
-    second=first+pbcDistance(first,second);
+  if(moldat && moldat->isWhole()) {
+    // make sure the tree has been constructed
+    if(!tree) tree=std::make_unique<Tree>(moldat);
+    const auto & tree_indexes=tree->getTreeIndexes();
+    const auto & root_indexes=tree->getRootIndexes();
```

Comment:
I may have read the Tree class wrong:
It appears to me that `Tree::getTreeIndexes();` and `Tree::getRootIndexes();` are getters that return a copy to the respective internal variable in the tree.

It seems to me that that `const auto&` creates a reference to the copy that is returned by the getter
I think [this example](https://coliru.stacked-crooked.com/a/f10d9aab881ff779) shows better what I am saying.

It may be a good idea to the getters return a `const reference`, so you will have a read-only accessor without any copy.

Reply:
You are right, I've already seen this (the original version was returning by copy as well) and I agree it should be fixed. Thanks!